bin/lib/rust_library_builder.py

```diff
@@ -0,0 +1,453 @@
+import contextlib
+import glob
+import hashlib
+import itertools
+import json
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+from collections import defaultdict
+from enum import Enum, unique
+from pathlib import Path
+from typing import Dict, Any, List, Optional, Generator, TextIO
+
+import requests
+
+from lib.amazon import get_ssm_param
+from lib.amazon_properties import get_properties_compilers_and_libraries
+from lib.library_build_config import LibraryBuildConfig
+
+build_supported_os = ['Linux']
+build_supported_buildtype = ['Debug']
+build_supported_arch = ['x86_64']
+build_supported_stdver = ['']
+build_supported_stdlib = ['']
+build_supported_flags = ['']
+build_supported_flagscollection = [['']]
+
+_propsandlibs: Dict[str, Any] = defaultdict(lambda: [])
+
+GITCOMMITHASH_RE = re.compile(r'^(\w*)\s.*')
+CONANINFOHASH_RE = re.compile(r'\s+ID:\s(\w*)')
+
+
+@unique
+class BuildStatus(Enum):
+    Ok = 0
+    Failed = 1
+    Skipped = 2
+    TimedOut = 3
+
+
+build_timeout = 600
+
+conanserver_url = "https://conan.compiler-explorer.com"
```

Comment:
I see you updated conan to support `rust` - presumably this is purely to track the build statuses etc?

Reply:
this url is indeed just for success/failure + logging + commithash annotation